src/kbmod/search/kernels.cu

```diff
@@ -139,61 +140,20 @@ __global__ void searchFilterImages(int imageCount, int width, int height,
             (params.do_sigmag_filter && currentT.lh < params.minLH))
             continue;
 
+        // If we are doing on GPU filtering, run the sigmaG filter
+        // and recompute the likelihoods.
         if (params.do_sigmag_filter) {
-            // Sort the the indexes (idxArray) of lcArray in ascending order.
-            for (int j = 0; j < imageCount; j++) {
-                for (int k = j + 1; k < imageCount; k++) {
-                     if (lcArray[idxArray[j]] > lcArray[idxArray[k]]) {
-                         tmpSortIdx = idxArray[j];
-                         idxArray[j] = idxArray[k];
-                         idxArray[k] = tmpSortIdx;
-                     }
-                }
-            }
-
-            // Compute index of the three percentile values in lcArray
-            // from the given bounds sGL0, 0.5 (median), and sGL1.
             int minKeepIndex = 0;
-            int maxKeepIndex = imageCount - 1;
-            int imgCountPlus1 = imageCount + 1;
-            const int percentiles[3] = {
-                int(imgCountPlus1 * params.sGL_L + 0.5) - 1,
-                int(imgCountPlus1 * 0.5 + 0.5) - 1,
-                int(imgCountPlus1 * params.sGL_H + 0.5) - 1};
-
-            // Compute the lcValues that at +/- 2*sigmaG from the median.
-            // This will be used to filter anything outside that range.
-            float sigmaG = params.sigmaGCoeff * (lcArray[idxArray[percentiles[2]]]
-                               - lcArray[idxArray[percentiles[0]]]);
-            float minValue = lcArray[idxArray[percentiles[1]]] - 2 * sigmaG;
-            float maxValue = lcArray[idxArray[percentiles[1]]] + 2 * sigmaG;
-
-            // Find the index of the first value in lcArray greater
-            // than or equal to minValue.
-            for (int i = 0; i <= percentiles[1]; i++) {
-                int idx = idxArray[i];
-                if (lcArray[idx] >= minValue) {
-                    minKeepIndex = i;
-                    break;
-                }
-            }
-            
-            // Find the index of the last value in lcArray less
-            // than or equal to maxValue.
-            for (int i = percentiles[1] + 1; i < imageCount; i++) {
-                int idx = idxArray[i];
-                if (lcArray[idx] <= maxValue) {
-                    maxKeepIndex = i;
-                } else {
-                    break;
-                }
-            }
-            
+            int maxKeepIndex = num_seen - 1;
+            sigmaGFilteredIndicesCU(lcArray, num_seen, params.sGL_L, params.sGL_H,
```

Comment:
nice, I was wondering why we had separate sigma filtering code when we already had a function for it on GPU 👍 this cleans stuff up a lot.